contrib/util/dupscore.cli.php

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/php
+<?php
+
+/**
+ * CLI script to compute patient duplication scores in patient_data.dupscore.
+ * The score is a measure of the likelihood that the patient is a duplicate of
+ * some patient created before it. Optional arguments specifying values are:
+ *
+ * --webdir   The full path to the OpenEMR web directory. Defaults to the directory
+ *            two levels above that of this script.
+ * --site     The site ID. Defaults to "default".
+ * --maxmins  The maximum number of minutes to run. Defaults to 60. Use 0 for no limit.
+ *
+ * Arguments not having a value may be:
+ *
+ * -q         Suppresses messages on stdout.
+ * -c         Clears existing scores to recompute all of them; except scores of -1
+ *            are not cleared because they are manually assigned.
+ *
+ * Because we are comparing every patient with every other patient, this script can
+ * run for a very long time with a large database. Thus we want to do it offline.
+ * If --maxmins is exceeded the script will terminate but may be run again to resume
+ * where it left off.
+ *
+ * A common usage is:
+ * php /var/www/html/openemr/contrib/util/dupscore.cli.php --maxmins=240
+ *
+ * Here is a sample crontab entry to automatically run up to 2 hours nightly:
+ * 3 1 * * * root php /var/www/html/openemr/contrib/util/dupscore.cli.php -q --maxmins=120
+ *
+ * @package   OpenEMR
+ * @link      http://www.open-emr.org
+ * @author    Rod Roark <rod@sunsetsystems.com>
+ * @copyright Copyright (c) 2021 Rod Roark <rod@sunsetsystems.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
```

Comment:
for command line php script recommend placing exit at top (which can be uncommented when used) and ensuring only can be run via commandline. see here for example:
https://github.com/openemr/openemr/blob/master/contrib/util/ccda_import/import_ccda.php#L39-L42

Reply:
@bradymiller Would the test for php_sapi_name() be sufficient? That will ensure it's not being run via the web server. Since it only computes dupscore values it won't do a lot of damage if run accidentally from the command line.

Reply:
@sunsetsystems , sounds good. Agreed very benign script at this point.

Reply:
btw, this type of thing could also be run as a background service in OpenEMR (would push us to then make a mechanism in the background service for times of day when ok to run since wouldn't want to run this during the day).

Reply:
Yes I thought about the background service possibility but as you say it would need some work. The important thing is to run it after upgrading the site. New patients will have their dupscores computed at add time but those may become outdated if demographics of previous patients are fixed, which is why the script should run from time to time.